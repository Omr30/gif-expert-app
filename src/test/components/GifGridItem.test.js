import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
const { shallow } = require("enzyme");

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={url} /> )
    
    test('debe de mostrar el componente corractemente', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de tener un parrafo con el titulo', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    });
    
    test('debe de tener la imagen igual al url y alt ded los porps', () => {
        
        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);

    });
    
    test('debe tener animate__flipInX', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__flipInX') ).toBe( true );

    });    
    


})
