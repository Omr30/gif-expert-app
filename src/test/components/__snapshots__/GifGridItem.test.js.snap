// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente corractemente 1`] = `
<div
  className="card animate__animated animate__flipInX"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
